### PR TITLE
feat(js): fetch access token by authorization code

### PR DESCRIPTION
**Pull request by demonzoo**
## Summary

Fetch access token by authorization code

## Testing

- [x] Passed all unit tests

### CODE REVIEW

File: packages/js/src/core/fetch-token.test.ts [hunk 1]
```diff
@@ -1,19 +1,54 @@
+import { fetchTokenByAuthorizationCode } from '.';
```
Comment (IceHe) [inline packages/js/src/core/fetch-token.test.ts:1]:
```ts
import { fetchTokenByAuthorizationCode } from '.';
import { fetchTokenByRefreshToken, RefreshTokenTokenResponse } from './fetch-token';
```

better merge in one line?

File: packages/js/src/core/fetch-token.test.ts [hunk 2]
```diff
@@ -1,19 +1,54 @@
...
+      },
+      fetchFunction
+    );
+    expect(tokenResponse).toEqual(expectedTokenResponse);
+  });
+});
+
 describe('fetch access token by providing valid refresh token', () => {
-  test('get token response', async () => {
+  test('should return access token', async () => {
```
Comment (IceHe) [inline packages/js/src/core/fetch-token.test.ts:39]: should `return token` (more than access token).
Also, line 38.

image

File: packages/js/src/core/fetch-token.test.ts (hunk 1 above)
Comment (demonzoo) [reply packages/js/src/core/fetch-token.test.ts]: updated

File: packages/js/src/core/fetch-token.test.ts (hunk 2 above)
Comment (demonzoo) [reply packages/js/src/core/fetch-token.test.ts]: improved

File: packages/js/src/consts/index.ts [hunk 3]
```diff
@@ -1,3 +1,21 @@
...
+export enum TokenGrantType {
+  AuthorizationCode = 'authorization_code',
+  RefreshToken = 'refresh_token',
+}
+
+export enum QueryKey {
+  ClientId = 'client_id',
+  Code = 'code',
+  CodeVerifier = 'code_verifier',
+  ExpiresIn = 'expires_in',
```
Comment (simeng-li) [inline packages/js/src/consts/index.ts:14]: Do we have any expires_in query param?

**Review by simeng-li — APPROVED** [review]
Overall LGTM

File: packages/js/src/consts/index.ts (hunk 3 above)
Comment (demonzoo) [reply packages/js/src/consts/index.ts:14]: No, we don't. Good call.

**Pull request merged**